xapian-applications/omega/handler_libextractor.cc

```diff
@@ -0,0 +1,120 @@
+/** @file handler_libextractor.cc
+ * @brief Extract text and metadata using libarchive.
+ */
+/* Copyright (C) 2020 Parth Kapadia
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "handler.h"
+
+#include <extractor.h>
+
+#include <cstring>
+
+using namespace std;
+
+string title, keywords, author, pages;
```

Comment:
Not sure if using the same names for this variables is a good option as this practice can induce bugs. I think we should try to avoid it.

We can use an structure(as Olly've suggested) to store the necessary information and pass a pointer to it in `cls` instead of global variables.

Reply:
I have changed this.

Reply:
Should I avoid using the same variables inside the structure too? As I have done right now.

Reply:
No, that's not necessary. Only when two variables have the same name under a scope. This is not an error, but it can introduce bugs in the code (for example, if you forget :: when you want to refer to a global variable) and makes the code less readable.